hello_wifi/lib/hello_wifi.ex

```diff
@@ -27,21 +25,52 @@ defmodule HelloWiFi do
     Logger.info("[HelloWiFi] - WiFi Wizard stopped")
   end
 
-  defp maybe_start_wifi_wizard(_wifi_configured? = true) do
-    # By this point we know there is a wlan interface available
-    # and already configured. This would normally mean that you
-    # should then skip starting the WiFi wizard here so that
-    # the device doesn't start the WiFi wizard after every
-    # reboot.
-    #
-    # However, for the example we want to always run the
-    # WiFi wizard on startup. Comment/remove the function below
-    # if you want a more typical experience skipping the wizard
-    # after it has been configured once.
-    VintageNetWizard.run_wizard(on_exit: {__MODULE__, :on_wizard_exit, []})
+  defp maybe_start_wifi_wizard() do
+    with true <- has_wifi?() || :no_wifi,
+         true <- wifi_configured?() || :not_configured,
+         true <- has_networks?() || :no_networks do
+      # By this point we know there is a wlan interface available
+      # and already configured with networks. This would normally
+      # mean that you should then skip starting the WiFi wizard
+      # here so that the device doesn't start the WiFi wizard after
+      # every reboot.
+      #
+      # However, for the example we want to always run the
+      # WiFi wizard on startup. Comment/remove the function below
+      # if you want a more typical experience skipping the wizard
+      # after it has been configured once.
+      VintageNetWizard.run_wizard(on_exit: {__MODULE__, :on_wizard_exit, []})
+    else
+      :no_wifi ->
+        Logger.error(
+          "[#{inspect(__MODULE__)}] Device does not support WiFi - Skipping wizard start"
+        )
+
+      fault ->
```

Comment:
can fault be renamed to **status** throughout?

Reply:
Changed